Intermittent test - TestSpanReader_multiRead_followUp_query

The test `TestSpanReader_multiRead_followUp_query` is intermittently failing. I spotted this for the PR #1668, which isn't touching any Go code at all.

```
--- FAIL: TestSpanReader_multiRead_followUp_query (0.00s)
    reader_test.go:279: 
        	Error Trace:	reader_test.go:279
        	            				reader_test.go:105
        	            				reader_test.go:209
        	Error:      	Not equal: 
        	            	expected: &model.Span{TraceID:model.TraceID{Low:0x2, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef{}, Flags:0x0, StartTime:time.Time{wall:0x3b93dc90, ext:63700183028, loc:(*time.Location)(nil)}, Duration:0, Tags:[]model.KeyValue{}, Logs:[]model.Log{}, Process:(*model.Process)(0xc000141770), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
        	            	actual  : &model.Span{TraceID:model.TraceID{Low:0x1, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef{}, Flags:0x0, StartTime:time.Time{wall:0x3b93dc90, ext:63700183028, loc:(*time.Location)(nil)}, Duration:0, Tags:[]model.KeyValue{}, Logs:[]model.Log{}, Process:(*model.Process)(0xc0001417c0), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
        	            	
        	            	Diff:
        	Test:       	TestSpanReader_multiRead_followUp_query
    reader_test.go:280: 
        	Error Trace:	reader_test.go:280
        	            				reader_test.go:105
        	            				reader_test.go:209
        	Error:      	Not equal: 
        	            	expected: &model.Span{TraceID:model.TraceID{Low:0x1, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef{}, Flags:0x0, StartTime:time.Time{wall:0x3b93dc90, ext:63700183028, loc:(*time.Location)(nil)}, Duration:0, Tags:[]model.KeyValue{}, Logs:[]model.Log{}, Process:(*model.Process)(0xc000141360), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
        	            	actual  : &model.Span{TraceID:model.TraceID{Low:0x2, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef{}, Flags:0x0, StartTime:time.Time{wall:0x3b93dc90, ext:63700183028, loc:(*time.Location)(nil)}, Duration:0, Tags:[]model.KeyValue{}, Logs:[]model.Log{}, Process:(*model.Process)(0xc000141810), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}
        	            	
        	            	Diff:
        	Test:       	TestSpanReader_multiRead_followUp_query
```

--- a/plugin/storage/es/spanstore/reader_test.go
+++ b/plugin/storage/es/spanstore/reader_test.go
@@ -276,8 +276,11 @@ func TestSpanReader_multiRead_followUp_query(t *testing.T) {
 		require.NoError(t, err)
 		sModel2, err := toDomain.SpanToDomain(&spanID2)
 		require.NoError(t, err)
-		assert.EqualValues(t, traces[0].Spans[0], sModel1)
-		assert.EqualValues(t, traces[1].Spans[0], sModel2)
+
+		for _, s := range []*model.Span{sModel1, sModel2} {
+			found := reflect.DeepEqual(traces[0].Spans[0], s) || reflect.DeepEqual(traces[1].Spans[0], s)
+			assert.True(t, found, "span was expected to be within one of the traces but was not: %v", s)
+		}
 	})
 }
 

